bids/variables/io.py

```diff
@@ -205,6 +205,7 @@ def _load_time_variables(layout, dataset=None, columns=None, scan_length=None,
         except Exception as e:
             if scan_length is not None:
                 duration = scan_length
+                nvols = int(scan_length / tr)
```

Comment:
I think you want to round, not just cast...

```suggestion
                nvols = int(np.rint(scan_length / tr))
```